GitCommands/Git/AheadBehindDataProvider.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+using GitUIPluginInterfaces;
+using JetBrains.Annotations;
+
+namespace GitCommands.Git
+{
+    public interface IAheadBehindDataProvider
+    {
+        IDictionary<string, AheadBehindData> GetData(string branchName = "*");
+    }
+
+    public class AheadBehindDataProvider : IAheadBehindDataProvider
+    {
+        private readonly Func<IExecutable> _getGitExecutable;
+
+        private readonly Regex _aheadBehindRegEx =
+            new Regex(@"^\[(ahead (?<ahead>\d+))?(, )?(behind (?<behind>\d+))?\] (?<branch>.*)$",
+                RegexOptions.Compiled | RegexOptions.Multiline);
+
+        public AheadBehindDataProvider(Func<IExecutable> getGitExecutable)
+        {
+            _getGitExecutable = getGitExecutable;
+        }
+
+        public IDictionary<string, AheadBehindData> GetData(string branchName = "*")
+        {
+            return GetData(GitModule.SystemEncoding, branchName);
```

Comment:
I realised after I pushed it in that I did it incorrectly

```suggestion
            return GetData(null, branchName);
```